x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -273,109 +272,78 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
+			// providerPaymentStorage not found (this provider has no payments in this epoch and also no complaints) -> we need to add one complaint
 			emptyProviderPaymentStorageWithComplaint := types.ProviderPaymentStorage{
 				Index:                              providerStorageKey,
 				UniquePaymentStorageClientProvider: []*types.UniquePaymentStorageClientProvider{},
 				Epoch:                              epoch,
-				UnresponsivenessComplaints:         []string{clientAddr.String()},
+				ComplainersTotalCu:                 complainerCuToAdd,
 			}
-			k.SetProviderPaymentStorage(ctx, emptyProviderPaymentStorageWithComplaint)
-			continue
-		}
-		// providerPaymentStorage was found for epoch, start analyzing if unstake is necessary
-		// check if the same consumer is trying to complain a second time for this epoch
-		if slices.Contains(providerPaymentStorage.UnresponsivenessComplaints, clientAddr.String()) {
-			continue
-		}
-
-		providerPaymentStorage.UnresponsivenessComplaints = append(providerPaymentStorage.UnresponsivenessComplaints, clientAddr.String())
 
-		// now we check if we have more UnresponsivenessComplaints than maxComplaintsPerEpoch
-		if len(providerPaymentStorage.UnresponsivenessComplaints) >= maxComplaintsPerEpoch {
-			// we check if we have double complaints than previous "collectPaymentsFromNumberOfPreviousEpochs" epochs (including this one) payment requests
-			totalPaymentsInPreviousEpochs, err := k.getTotalPaymentsForPreviousEpochs(ctx, collectPaymentsFromNumberOfPreviousEpochs, epoch, chainID, sdkUnresponsiveProviderAddress)
-			totalPaymentRequests := totalPaymentsInPreviousEpochs + len(providerPaymentStorage.UniquePaymentStorageClientProvider) // get total number of payments including this epoch
-			if err != nil {
-				utils.LavaFormatError("lava_unresponsive_providers: couldnt fetch getTotalPaymentsForPreviousEpochs", err, nil)
-			} else if totalPaymentRequests*providerPaymentMultiplier < len(providerPaymentStorage.UnresponsivenessComplaints) {
-				// unstake provider
-				utils.LogLavaEvent(ctx, logger, types.ProviderJailedEventName, map[string]string{"provider_address": sdkUnresponsiveProviderAddress.String(), "chain_id": chainID}, "Unresponsive provider was unstaked from the chain due to unresponsiveness")
-				err = k.unSafeUnstakeProviderEntry(ctx, epochstoragetypes.ProviderKey, chainID, indexInStakeStorage, existingEntry)
-				if err != nil {
-					utils.LavaFormatError("unable to unstake provider entry (unsafe method)", err, &map[string]string{"chainID": chainID, "indexInStakeStorage": strconv.FormatUint(indexInStakeStorage, 10), "existingEntry": existingEntry.GetStake().String()})
-					continue
-				}
+			// add the new empty providerPaymentStorage to epochPayments
+			epochPayments, found, key := k.GetEpochPaymentsFromBlock(ctx, epoch)
+			if !found {
+				epochPayments = types.EpochPayments{Index: key, ClientsPayments: []*types.ProviderPaymentStorage{&emptyProviderPaymentStorageWithComplaint}}
```

Comment:
I can't because that function (AddEpochPayment) assumes that the providerPaymentStorage object holds a non-nil list of UniquePaymentStorageClientProvider. In other words, it doesn't support adding a providerPaymentStorage object without valid payments (I'm trying to append a providerPaymentStorage which only has complainerCU)